final class androidx.work.impl.utils.futures.AbstractFuture$Failure extends java.lang.Object
{
    static final androidx.work.impl.utils.futures.AbstractFuture$Failure FALLBACK_INSTANCE;
    final java.lang.Throwable exception;

    static void <clinit>()
    {
        androidx.work.impl.utils.futures.AbstractFuture$Failure$1 $r1;
        androidx.work.impl.utils.futures.AbstractFuture$Failure $r0;

        $r0 = new androidx.work.impl.utils.futures.AbstractFuture$Failure;

        $r1 = new androidx.work.impl.utils.futures.AbstractFuture$Failure$1;

        specialinvoke $r1.<init>("Failure occurred while trying to finish a future.");

        specialinvoke $r0.<init>($r1);

        androidx.work.impl.utils.futures.AbstractFuture$Failure.FALLBACK_INSTANCE = $r0;

        return;
    }

    void <init>(java.lang.Throwable)
    {
        java.lang.Throwable $r1;
        java.lang.Object $r2;
        androidx.work.impl.utils.futures.AbstractFuture$Failure r0;

        r0 := @this;

        $r1 := @parameter0;

        specialinvoke r0.<init>();

        $r2 = androidx.work.impl.utils.futures.AbstractFuture.checkNotNull($r1);

        $r1 = (java.lang.Throwable) $r2;

        r0.exception = $r1;

        return;
    }
}
